addons/test_mass_mailing/tests/test_performance.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from email.utils import formataddr
+
+from odoo.addons.test_mass_mailing.tests import common
+from odoo.tests.common import TransactionCase, users, warmup
+from odoo.tests import tagged
+from odoo.tools import mute_logger
+
+
+@tagged('mail_performance')
+class TestMassMailPerformance(common.MassMailingCase):
+
+    def setUp(self):
+        super(TestMassMailPerformance, self).setUp()
+        values = [{
+            'name': 'Recipient %s' % x,
+            'email_from': 'Recipient <rec.%s@example.com>' % x,
+        } for x in range(0, 100)]
```

Comment:
10 emails instead of 100 would exercise the code just as much, but I don't really think the overhead is significant in a complete test run, so why not.